be/src/vec/sink/vrow_distribution.cpp

```diff
@@ -456,15 +507,19 @@ Status VRowDistribution::generate_rows_distribution(
     }
 
     Status st = Status::OK();
-    if (_vpartition->is_auto_detect_overwrite()) {
+    if (_vpartition->is_auto_detect_overwrite() && !_deal_batched) {
```

Comment:
the code wired. seems the `deal_batched` should always be true in call func `_generate_rows_distribution_for_auto_overwrite`